server/v2/appmanager/service.go

```diff
@@ -52,13 +58,40 @@ type AppManager[T transaction.Tx] struct {
 
 	db store.Store
 
+	mempool mempool.Mempool[T]
+
 	lastBlockHeight *atomic.Uint64
 
 	initGenesis func(ctx context.Context, genesisBytes []byte) error
 
+	prepareHandler appmanager.PrepareHandler[T]
+
 	stf *stf.STF[T]
 }
 
+// BuildBlock builds a block when requested by consensus. It will take in a list of transactions and return a list of transactions
+func (a AppManager[T]) BuildBlock(ctx context.Context, block appmanager.BlockRequest, totalSize uint32) ([]T, error) {
```

Comment:
What is totalSize meant to indicate? Number of txs or total size of proposal (encoded)? I would make this clear in the godoc.

Reply:
total size is the amount of bytes that are free in a block, i can add it to the go doc

Reply:
Yes please 🙏